solarwinds_apm/traceoptions.py

```diff
@@ -31,28 +23,30 @@ class XTraceOptions:
     # pylint: disable=too-many-branches,too-many-statements
     def __init__(
         self,
-        context: typing.Optional[Context] = None,
+        xtraceoptions_header: str = "",
+        signature_header: str = "",
     ):
         """
         Args:
-          context: OTel context that may contain OTEL_CONTEXT_SW_OPTIONS_KEY,OTEL_CONTEXT_SW_SIGNATURE_KEY
+          xtraceoptions_header: extracted request header value
+          signature_header: extracted request header value
         """
         self.ignored = []
         self.options_header = ""
-        self.signature = None
+        self.signature = ""
         self.custom_kvs = {}
         self.sw_keys = ""
         self.trigger_trace = 0
         self.timestamp = 0
 
-        if not context:
-            return
-        options_header = context.get(INTL_SWO_X_OPTIONS_KEY, "")
-        # store original header for sample decision later
-        self.options_header = options_header
+        if signature_header:
+            self.signature = signature_header
+            # store original header for c-lib sample decision later
+            # only if signature_header exists
+            self.options_header = xtraceoptions_header
```

Comment:
This was Raph's nice suggestion to save on space because options_header [only comes in play for a signed TT request](https://github.com/librato/solarwinds-apm-liboboe/blob/master/liboboe/settings.cpp#L1046-L1054).